bf2fc6cc711aee1a0c2a/kas-fleet-manager
# feat: replace dynamic scale up mechanism

## Description

Related to https://issues.redhat.com/browse/MGDSTRM-9133, https://issues.redhat.com/browse/MGDSTRM-9138.

The old dynamic scale up logic where clusters were automatically created
in a provider's region when there were no other clusters has been replaced
by the new dynamic scale up logic which takes into account more scenarios.
Some integration tests have been refactored to make them pass and prevent
unintended data plane cluster scale ups.

## Verification Steps

Run integration tests against a real OCM environment and check that no more than 1 data plane cluster is created.
This can be checked with:
```
ocm list clusters --parameter search="name like '%mk-%'"
```
and observe how many were created by the user running kfm since the tests execution.
Another way is to check the generated logs and look for the `registering new data plane` text. There should be one appearance only.


## Checklist (Definition of Done)
